### PR TITLE
[woptim] Add support for poodle machine

### PR DESCRIPTION
**Pull request by adrienbernede**
### Summary

Add poodle to the set of CI machines.

### CODE REVIEW

**Comment by rhornung67** [comment]
@adrienbernede I merged your mpi-related PR. Now this conflicts with recent changes to radiuss-spack-configs. I tried to manually resolve them locally by merging develop into into the `woptim/poodle` branch. The `git merge` command tells me there are unmerged files, but not which files have conflicts. `git status` gives me only this:

`Unmerged paths:
  (use "git add <file>..." to mark resolution)

	both modified:   scripts/radiuss-spack-configs`

Not sure how to fix this...

**Comment by adrienbernede** [comment]
@rhornung67 my bad, I’m gonna clear this out.

**Comment by adrienbernede** [comment]
@rhornung67 here is my mistake: I pushed for #1557 to be merged before merging the changes in RADIUSS Spack Configs. Since we had 2 branches in RAJA with diverging changes is RADIUSS Spack Configs, there was a conflict.

Conflict resolution: merge both changes in RADIUSS Spack Configs @ main, and get RAJA to point at that new main reference.

**Review by rhornung67 — APPROVED** [review]
Hooray! Testing on poodle. Thanks @adrienbernede

File: .gitlab/jobs/poodle.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+############
+# Extra jobs
+############
+# We do not recommend using ${PROJECT_<MACHINE>_VARIANTS} and
+# ${PROJECT_<MACHINE>_DEPS} in the extra jobs. There is no reason not to fully
+# describe the spec here.
+
+clang_14_0_6_openmp_off:
+  variables:
+    SPEC: " ~shared ~openmp +tests %clang@14.0.6"
```
Comment (MrBurmark) [inline .gitlab/jobs/poodle.yml:47]: Why is this ~openmp while others are +openmp?

File: scripts/gitlab/build_and_test.sh [hunk 2]
```diff
@@ -144,7 +144,7 @@ then
     echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
 
     # Map CPU core allocations
-    declare -A core_counts=(["lassen"]=40 ["ruby"]=28 ["corona"]=32 ["rzansel"]=48 ["tioga"]=32)
+    declare -A core_counts=(["lassen"]=40 ["ruby"]=28 ["poodle"]=28 ["corona"]=32 ["rzansel"]=48 ["tioga"]=32)
```
[the remark this answers is not in the document]
Comment (rhornung67) [reply scripts/gitlab/build_and_test.sh:147]: according to the Intel spec sheet the CPUs on poodle have 52 cores each: https://www.intel.com/content/www/us/en/products/sku/231748/intel-xeon-platinum-8470n-processor-97-5m-cache-1-70-ghz/specifications.html

However, the LC platform page says each node has 112 cores: https://hpc.llnl.gov/hardware/compute-platforms

Not sure which is correct...

**Review by artv3 — APPROVED** [review]

File: scripts/gitlab/build_and_test.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (MrBurmark) [reply scripts/gitlab/build_and_test.sh:147]: That's what I'm seeing via hwloc.

[the remark this answers is not in the document]
Comment (rhornung67) [reply scripts/gitlab/build_and_test.sh:147]: I haven't looked. Does it show two sockets?

[the remark this answers is not in the document]
Comment (rhornung67) [reply scripts/gitlab/build_and_test.sh:147]: I'm not clear on what role those numbers in the test script play in the CI testing.

[the remark this answers is not in the document]
Comment (MrBurmark) [reply scripts/gitlab/build_and_test.sh:147]: Ya there are 2 sockets.

File: .gitlab/jobs/poodle.yml (hunk 1 above)
Comment (adrienbernede) [reply .gitlab/jobs/poodle.yml:47]: Hmm, I wouldn’t be able to track the history of it, but it’s not a typo, the name of the job is coherent.

I would say, I you want it to be "+openmp", we’ll make it that way :D

File: scripts/gitlab/build_and_test.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (adrienbernede) [reply scripts/gitlab/build_and_test.sh:147]: I like that you’ve looked into the details.

There is no CI reservation on poodle (yet), so I’m using the pdebug queue. This queue is quite limited on poodle: 1 node for 30min. So I allowed the builds to run concurrently on the allocation and limited the number of threads to get a better efficiency in that context. The result is that the jobs fit, while with 56 they did not.

In theory, the number here should be either 56 or 112 (I don’t know the exact configuration of that machine, I inferred 2 sockets of 56 cores.)

[the remark this answers is not in the document]
Comment (adrienbernede) [reply scripts/gitlab/build_and_test.sh:147]: (posting the comment updated the conversation... now I have all the previous answers...)

[the remark this answers is not in the document]
Comment (adrienbernede) [reply scripts/gitlab/build_and_test.sh:147]: @rhornung67 these are for the number of threads used in the build (cmake command) you can have has many as you want.

Each CI job is a sub allocation in a shared allocation. There are 2 cases:
- jobs run concurrently on the node
- jobs run sequentially on the node

You explicitly asked that job run sequentially for RAJA in the past. However in poodle case, I don’t have enough flexibility with the parent allocation to get the CI to fit sequentially. Instead I am packing them all in concurrent mode, which explains why I am reducing the thread concurrency for each job.

I asked for the CI reservation to be set on poodle to improve things.

**Comment by adrienbernede** [comment]
@MrBurmark I’m gonna merge it to get https://github.com/LLNL/RAJAPerf/pull/332 in a clean state. If you are not satisfied with my answers, we can make the appropriate changes in a subsequent PR.

**Pull request merged**